testsuite/tests/typing-layouts-arrays/test_int_or_null_array.ml

```diff
@@ -0,0 +1,105 @@
+(* TEST
+ include stdlib_stable;
+ include stdlib_upstream_compatible;
+ readonly_files =
+   "gen_u_array.ml test_gen_u_array.ml gen_product_array_helpers.ml";
+ modules = "${readonly_files}";
+ flambda2;
+ stack-allocation;
+ {
+   bytecode;
+ }
+ {
+   native;
+ }
+*)
+
+open Gen_product_array_helpers
+open Stdlib_stable
+open Stdlib_upstream_compatible
+
+(* If copying this test for a new shape, you should only have to
+   change the bit between here and the next comment. See README.md in this
+   test directory. *)
+
+(* CR or-null: separability isn't inferred without the signature, investigate. *)
```

Comment:
This is strange indeed. Here's a small repro:
```ocaml
type t = int or_null

module type S = sig
  type t : any mod separable
end

module type S' = S with type t = t
```
Can you make a ticket for this, if you haven't?